### ISSUE TITLE
Crash: Payload may not be larger than 32767 bytes

### ISSUE THREAD

**Issue by KirbyQK** [CoFH/Feedback#865]
I've got limited coding knowledge so I can't be sure it is an issue with TE, but after reading around on what other users have done to encounter this problem, I believe it is storage related.

I got this crash when putting a bee into a strongbox. Roloading the world, opening the strongbox would immediately cause this crash. I was able to pull bees out with an itemduct and servo, and even just having one less of any bee in the strongbox was completely fine.

This is what I'm looking at:

![image](https://user-images.githubusercontent.com/35182387/35097487-75fe40a2-fca4-11e7-89b2-0a2f90620b39.png)

Moving any additional bee into the strongbox reproduces the crash without fail.

This is all that seems to be reported in the log;

`[22:52:12] [Netty Local Client IO #3/ERROR] [FML]: NetworkDispatcher exception
java.lang.IllegalArgumentException: Payload may not be larger than 32767 bytes
	at net.minecraft.network.play.client.CPacketCustomPayload.<init>(CPacketCustomPayload.java:28) ~[lh.class:?]
	at net.minecraftforge.fml.common.network.internal.FMLProxyPacket.toC17Packet(FMLProxyPacket.java:142) ~[FMLProxyPacket.class:?]
	at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher.write(NetworkDispatcher.java:534) ~[NetworkDispatcher.class:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) ~[AbstractChannelHandlerContext.class:4.1.9.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:801) ~[AbstractChannelHandlerContext.class:4.1.9.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:814) ~[AbstractChannelHandlerContext.class:4.1.9.Final]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:794) ~[AbstractChannelHandlerContext.class:4.1.9.Final]
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:831) ~[AbstractChannelHandlerContext.class:4.1.9.Final]
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1032) ~[DefaultChannelPipeline.class:4.1.9.Final]
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:296) ~[AbstractChannel.class:4.1.9.Final]
	at net.minecraft.network.NetworkManager$4.run(NetworkManager.java:245) [gw$4.class:?]
	at io.netty.channel.DefaultEventLoop.run(DefaultEventLoop.java:54) [DefaultEventLoop.class:4.1.9.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) [SingleThreadEventExecutor$5.class:4.1.9.Final]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]`

I hope that's enough info, thanks!

**Comment by covers1624** [comment]
Yep.. Not sure of the best way around this, basically the packet is too large due to all the bee's NBT tags, iirc `getNbtShareTag` was for this exact purpose.

**Comment by KirbyQK** [comment]
Not sure if it makes any difference, but crates from Actually Additions seem to work ok;

![image](https://user-images.githubusercontent.com/35182387/35099006-7b8317ae-fcaa-11e7-89d7-0e1127114ca3.png)

**Comment by KingLemming** [comment]
So what's interesting is that I've tried to make this happen in the past and absolutely could not. I'm going to try again, and I bet I still won't be able to.

**Comment by KirbyQK** [comment]
It's probably worth mentioning that all of the bees shown in the first screenshot are the direct result of breeding, so they probably have a bunch more data attached to them than normal bees anyway. They've all been analyzed as well. I decided to test this further on my save and moved all of the bees into the crate I snipped above. No crash!

![image](https://user-images.githubusercontent.com/35182387/35124952-18455f10-fcfc-11e7-9761-3065c49acfdc.png)

Top Google result I found was for iron backpacks, which is EXTREMELY similar, almost identical; https://github.com/gr8pefish/IronBackpacks/issues/117

Another decent result was this; https://github.com/PrinceOfAmber/Storage-Network/issues/38

The TL;DR is that pulling too many different types of glass/wood etc. at once for a recipe would cause this crash, their fix/changelog is here; https://github.com/PrinceOfAmber/Storage-Network/pull/45/files#diff-2103efac9db966ed8688bc9b541c1638R45

I hope that helps, I'm happy to give you my save file if that's of any use, I'm on the DW20 1.12 pack, version 1.4.0

**Comment by KirbyQK** [comment]
Just to add, it looks like Direwolf20 himself ran into this crash a little while ago: https://youtu.be/mcfb5RmJF2A?t=13m59s

**Comment by KingLemming** [comment]
So the problem with this is that we aren't sending the packet. Item sync is handled by vanilla code. This is either a raw vanilla issue or it's something Forge is causing by trying to be clever.

**Comment by malte0811** [comment]
I debugged this for IE recently (BluSunrize/ImmersiveEngineering#2756). Turns out forestry doesn't implement `getNBTShareTag`, so inventories can't know how much of the NBT they actually need to send. With a proper implementation of that method the vanilla/forge sync code works perfectly, even if the server-side NBT is so large it causes a chunk saving failure.
I'd be interested to know why some other inventories work though, any ideas?

**Comment by KingLemming** [comment]
Interesting. So did you have a fix, or is this just something we punt to Forestry?

**Comment by malte0811** [comment]
I'd just say it's forestry's fault. I'm not sure whether it would be feasible to implement custom syncing for this, but it probably isn't. There's a comment from a forestry dev in the issue I linked, seems like they aren't going to fix this in 1.12 since it would break addons. It might be worth looking for differences between AA's crates (since they don't seem to crash) and IE's crates/TE's strongboxes, but it might just be that the NBT of the bees that were put into the AA crate happened to be easier to compress than that in the strongbox.

**Comment by temp1011** [comment]
Relevant issue https://github.com/ForestryMC/ForestryMC/issues/561. Should get fixed at some point.
